src/pages/Dashboard/Dashboard.tsx

```diff
@@ -21,14 +22,19 @@ import { WalletDashboard } from './WalletDashboard'
 const direction: StackDirection = { base: 'column', xl: 'row' }
 const maxWidth = { base: 'full', lg: 'full', xl: 'sm' }
 
-const dashboardHeader = <DashboardHeader />
-
 export const Dashboard = memo(() => {
   const translate = useTranslate()
   const isDefiDashboardEnabled = useFeatureFlag('DefiDashboard')
   const { path } = useRouteMatch()
   const isNftsEnabled = useFeatureFlag('Jaypegz')
 
+  const isOpportunitiesLoading = useFetchOpportunities()
+
+  const dashboardHeader = useMemo(
+    () => <DashboardHeader isOpportunitiesLoading={isOpportunitiesLoading} />,
```

Comment:
Feels like we should be able to call this hook directly from `<DashboardHeader />` and leverage the loading state from within? This feels like a weird tight coupling IMO

Reply:
Looking more closely, it is here because there are multiple pages/components underneath that need this data, though I guess the dashboard header is the only one that needs loading state

Reply:
Actually yeah the total portfolio balance depends on these to be loaded as well as the Defi and Rewards pages themselves. Wanted to have it called once here to prevent duplicate fetches until we can optimize the queries themselves.